pkg/podrescompat/client.go

```diff
@@ -0,0 +1,83 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podrescompat
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+
+	podresourcesapi "k8s.io/kubelet/pkg/apis/podresources/v1"
+
+	"github.com/openshift-kni/resource-topology-exporter/pkg/sysinfo"
+)
+
+type sysinfoClient struct {
+	sysInfoConfig string
+	cli           podresourcesapi.PodResourcesListerClient
+}
+
+func NewSysinfoClientFromLister(cli podresourcesapi.PodResourcesListerClient, sysInfoConfig string) podresourcesapi.PodResourcesListerClient {
+	return &sysinfoClient{
+		cli:           cli,
+		sysInfoConfig: sysInfoConfig,
+	}
+}
+
+func (sc *sysinfoClient) List(ctx context.Context, in *podresourcesapi.ListPodResourcesRequest, opts ...grpc.CallOption) (*podresourcesapi.ListPodResourcesResponse, error) {
+	return sc.cli.List(ctx, in, opts...)
+}
+
+func (sc *sysinfoClient) GetAllocatableResources(ctx context.Context, in *podresourcesapi.AllocatableResourcesRequest, opts ...grpc.CallOption) (*podresourcesapi.AllocatableResourcesResponse, error) {
+	resp, err := sc.cli.GetAllocatableResources(ctx, in, opts...)
+	if err != nil {
```

Comment:
I would check if the error is feature-gate related, otherwise, I would just return with the error (but I'm not sure if there is a way to know that)

Reply:
the only reason why the kube call may fail is because of the feature gate being disabled: https://github.com/kubernetes/kubernetes/blob/release-1.21/pkg/kubelet/apis/podresources/server_v1.go#L79
other than that, unfortunately the podresourcesapi errors are not standardized (yet?) and I'd really really really like to avoid to make logic on the error text, unless we want that very badly.